src/app/data.service.spec.ts

```diff
@@ -1,12 +1,25 @@
 import { TestBed } from '@angular/core/testing';
 
 import { DataService } from './data.service';
+import {HttpClient, HttpClientModule, HttpHandler, HttpResponse} from '@angular/common/http';
+import {HttpTestingController} from '@angular/common/http/testing';
 
 describe('DataService', () => {
-  beforeEach(() => TestBed.configureTestingModule({}));
 
-  it('should be created', () => {
-    const service: DataService = TestBed.get(DataService);
-    expect(service).toBeTruthy();
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientModule],
+      providers: [DataService]
+    });
   });
+  const dataService: DataService = TestBed.get(DataService);
+  const httpMock: HttpTestingController = TestBed.get(HttpTestingController);
+
+
+  it('should connect to the database', () => {
+    dataService.checkHttpConnection().subscribe();
+    const req = httpMock.expectOne('http://localhost:8002/api/', 'call to api');
```

Comment:
this should not be hardcoded.. when the API server changes tests will break. This should be stored in a config property file and imported into tests so if it changes, it will be only called in one place!